src/libraries/System.Private.CoreLib/src/System/Type.cs

```diff
@@ -526,6 +526,19 @@ public virtual Array GetEnumValues()
             throw NotImplemented.ByDesign;
         }
 
+        [UnconditionalSuppressMessage("AotAnalysis", "IL3050:RequiresDynamicCode",
```

Comment:
It's probably okay for arrays of primitive types (we'll likely have code for them) but in general this code can still be reachable if someone implements their own System.Type. RuntimeType is not the only implementation. We cannot assume everyone is going to override in the suppression in general.

Reply:
We can change the base implementation to `throw NotImplemented.ByDesign;` instead and get rid of the warning suppression here. Do you think that it would be better?

Reply:
We could also do a switch over `Type.GetTypeCode` and `new` the array types using static code.

Reply:
It won't allow us to get rid the warning suppression. The method is still going to call `GetEnumValues` that will trigger the AOT compatibility warning.

Reply:
Could we just pull over the whole implementation from [RoType.cs](https://github.com/dotnet/runtime/pull/73057/files#diff-7845dfb1c2c81d46adb70b62972bf2de618aa1818af6eab16dbbbf6179c6f746) in this pull request to Type.cs?

Reply:
I have looked at the prior art. We have prior art of both:
- Throwing `NotSupportedException(SR.NotSupported_SubclassOverride)` for new methods. For example, we have done it for `IsByRefLikeImpl`. We could have implemented polyfill using custom attribute APIs, but we did not.
- Trying to provide slow polyfill. For example, we have done it for `GetMemberWithSameMetadataDefinitionAs`.

I think that I like the throwing option better. Less code in the runtime and there are not that many custom reflection object models that matter to make it their problem.